terraform/protoc-gen-terraform-teleport.yaml

```diff
@@ -134,6 +134,24 @@ exclude_fields:
     - "UserSpecV2.CreatedBy"
     - "UserSpecV2.Status"
 
+    # ClusterNetworkingConfig excluded because we currently do not support oneOf directives
+    - "ClusterNetworkingConfigV2.Spec.TunnelStrategy"
+
+name_overrides:
+    # Context: https://github.com/gravitational/teleport/pull/12054
+    # We created a new field `RedirectURLs` but its jsontag value is `redirect_url`
+    # `redirect_url` already existed as `RedirectURL` and its jsontag is now `-`
+    # The generator creates two fields with the same name:
+    # - RedirectURLs has jsontag `redirect_url`, so it creates an entry for `redirect_url`
+    # - RedirectURL has jsontag `-`, so it converts the name into snake_case, creating another entry for `redirect_url`
+    #
+    # The solution applied here is to rename `RedirectURLs` from `redirect_url` to `redirect_urls`
+    # This way we don't break configurations with a minor release.
+
+    # At v10 - we will change the type of the `redirect_url` from string to list of strings <- we break user's configuration here
```

Comment:
> we break user's configuration here

Could you clarify this, what will break?

How does this change affect terraform users that have existing OIDC connectors or create new ones?

Reply:
Terraform does not allow for a field to have two types.
So, in this case, the field `redirect_url` in terraform must be a string **or** a list of strings.

As of `v<9.3.5`, its type is a String and we can't change that in a minor release.

[Teleport API 9.3.5 added](https://github.com/gravitational/teleport/pull/13046) a new field that we also must support: RedirectURLs
So, we added a new field: `redirect_urls`

So, with this PR (v9.3.5+) we have:
```
redirect_url: <string> # maps to RedirectURL
redirect_urls: <list of strings> # maps to RedirectURLs
```

The plan is to change `redirect_url`'s type from string to list of strings at most at v11.0.0 (I previously said v10)
Why v11? That's when the RedirectURL field is going away.
When it happens, users that were still using `redirect_url` field, must change it 
```tf
redirect_url: "https://example.org/redirect"
```
to
```tf
redirect_url: ["https://example.org/redirect"]
```

I tried to reword the explanation in the file
Please let me know if it makes sense

Reply:
> Terraform does not allow for a field to have two types.
So, in this case, the field redirect_url in terraform must be a string or a list of strings.

In this case, I don't think we should aim for the end result of:
```tf
redirect_url: ["https://example.org/redirect"]
```

Can we instead support the following, with no further deprecation steps necessary:
```
redirect_url: <string> # maps to RedirectURLs[0]
redirect_urls: <list of strings> # maps to RedirectURLs - takes precedence over redirect_url
```